nb/projects/BarbourSmith/belt_sprockets/belt_sprockets.nb

```diff
@@ -27,15 +27,14 @@ const additionalToothDepth = 0; //mm
 
 // Functions
 
-const ToothSpaceingCurvefit = (teeth, b, c, d) => {
-  return ((c * Math.pow(teeth, d)) / (b + Math.pow(teeth, d))) * teeth;
-};
+const ToothSpaceingCurvefit = (teeth, b, c, d) =>
+  ((c * Math.pow(teeth, d)) / (b + Math.pow(teeth, d))) * teeth;
 
 const ToothSpacing = (teeth, tooth_pitch, pitch_line_offset) =>
   2 * ((teeth * tooth_pitch) / (3.14159265 * 2) - pitch_line_offset);
 
 const ToothProfileToFace = (...points) =>
-  Shape.fromOpenPath(points).loop().fill();
+  Shape.chain(Shape.fromOpenPath(points)).loop().fill();
```

Comment:
Chain is nice here

Reply:
This is more of a hack to avoid thinking about how to fix it more cleanly. :)

The problem is that Loop doesn't support receiving raw coordinates, but fromOpenPaths does.

Need to think about how to rationalize all of that as well -- the parameter conversion stuff is a bit too magical.

Reply:
How does the parameter conversion stuff work as it is?

Reply:
toShape is the most common conversion -- if it receives a function it will apply the current shape to it.
toShapes is similar, but will handle Group, and Array.
toCoordinate tries to produce [x, y, z], and so on.

Those are reasonably straight-forward.

Some others probably need rethinking, like toNestedValues, etc.